Gems/ROS2/Code/Source/Camera/CameraSensor.h

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
+
+#include "ROS2/ROS2GemUtilities.h"
+#include <Atom/Feature/Utils/FrameCaptureBus.h>
+#include <chrono>
+#include <rclcpp/publisher.hpp>
+#include <sensor_msgs/msg/image.hpp>
+#include <std_msgs/msg/header.hpp>
+
+class Entity;
+
+namespace ROS2
+{
+
+    //! Structure containing all information required to create the camera sensor
+    struct CameraSensorDescription
+    {
+        //! Constructor to create the description
+        //! @param cameraName - name of the camera; used to differentiate cameras in a multi-camera setup
+        //! @param verticalFov - vertical field of view of camera sensor
+        //! @param width - camera image width in pixels
+        //! @param height - camera image height in pixels
+        CameraSensorDescription(const AZStd::string& cameraName, float verticalFov, int width, int height);
+
+        const float m_verticalFieldOfViewDeg; //!< camera vertical field of view
+        double m_verticalFieldOfViewRad;
+        const int m_width; //!< camera image width in pixels
+        const int m_height; //!< camera image height in pixels
+        const AZStd::string m_cameraName; //!< camera name to differentiate cameras in a multi-camera setup
+
+        const float m_aspectRatio; //!< camera image aspect ratio; equal to (width / height)
+        const AZ::Matrix4x4 m_viewToClipMatrix; //!< camera view to clip space transform matrix; derived from other parameters
+        const AZStd::array<double, 9> m_cameraIntrinsics; //!< camera intrinsics; derived from other parameters
+
+    private:
+        AZ::Matrix4x4 MakeViewToClipMatrix() const;
+
+        AZStd::array<double, 9> MakeCameraIntrinsics() const;
+
+        void validateParameters() const;
```

Comment:
This function name should use UpperCamelCase for the name per the Coding Standard: https://github.com/o3de/sig-core/blob/main/governance/Coding-Standards-and-Style-Guide.md#function-names